src/Tizen.System/PowerUsage/PowerUsage.cs

```diff
@@ -0,0 +1,217 @@
+        Gps = 5
+    }
+
+    /// <summary>
+    /// Provides information related to the power consumption by applications or by hardware resources on a battery-powered device for a given duration of time.
+    /// </summary>
+    /// <feature>http://tizen.org/feature/battery</feature>
+    /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+    /// <since_tizen> 7 </since_tizen>
+    public static class PowerUsage
+    {
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="rtypes">list of resource type identifiers like BLE, WiFi, CPU etc.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        /// <exception cref="ArgumentException">When an invalid parameter value is set.</exception>
+        /// <exception cref="UnauthorizedAccessException">If the privilege is not set.</exception>
+        /// <exception cref="InvalidOperationException">In case of any system error.</exception>
+        /// <exception cref="NotSupportedException">In case power usage is not supported</exception>
+        public static IDictionary<PowerUsageResourceType, double> GetPowerUsage(string appID, IList<PowerUsageResourceType> rtypes, DateTime start, DateTime end)
+        {
+            IntPtr dataHandle = IntPtr.Zero;
+            DateTimeOffset startTime = DateTime.SpecifyKind(start, DateTimeKind.Utc);
+            DateTimeOffset endTime = DateTime.SpecifyKind(end, DateTimeKind.Utc);
+            IDictionary<PowerUsageResourceType, double> batteryUsage = new Dictionary<PowerUsageResourceType, double>();
+            try
+            {
+                PowerUsageError ret = (PowerUsageError)Interop.PowerUsage.GetPowerUsageByAppForAllResources(appID, startTime.ToUnixTimeSeconds(), endTime.ToUnixTimeSeconds(), out dataHandle);
+                if (ret != PowerUsageError.None)
+                {
+                    Log.Error(PowerUsageErrorFactory.LogTag, "Error getting battery usage for all resources" + ret);
+                    throw PowerUsageErrorFactory.ThrowPowerUsageException(ret);
+                }
+
+                if (rtypes != null)
+                {
+                    foreach (PowerUsageResourceType type in rtypes)
+                    {
+                        batteryUsage.Add(type,GetPowerUsage(dataHandle, type));
```

Comment:
To avoid duplication case
```suggestion
                        batteryUsage[type] = GetPowerUsage(dataHandle, type));
```